core-api/src/main/java/com/sequenceiq/cloudbreak/cloud/MaintenanceVersionComparator.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.cloudbreak.cloud;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * This comparator supports only the three-part version number format like 1.2.3
+ */
+@Component
+public class MaintenanceVersionComparator {
```

Comment:
According to the name of this class, when we eventually will support minor and major upgrades, new comparators will be needed for those, which smells like code duplication. I would advise to extend the already existing Version class with segment-specific comparison reusing parts of the code there.